src/Compilers/CSharp/Portable/Symbols/Source/SourceMemberFieldSymbol.cs

```diff
@@ -571,5 +571,12 @@ internal override bool IsDefinedInSourceTree(SyntaxTree tree, TextSpan? definedW
 
             return false;
         }
+
+        internal override void AfterAddingTypeMembersChecks(ConversionsBase conversions, DiagnosticBag diagnostics)
+        {
+            bool includeNullability = DeclaringCompilation.IsFeatureEnabled(MessageID.IDS_FeatureNullableReferenceTypes);
```

Comment:
>IsFeatureEnabled [](start = 59, length = 16)

This seems wrong. During binding, we shouldn't use `conversions.WithNullability(true)`. We should just use `conversions` (which has the nullability flag set to `false`). 
Tagging @cston to confirm

Reply:
I had a feeling this was not right--it's not granular enough to just check the Compilation--but I believe something is required here to handle nullability. See the below failures when I remove the `conversions.WithNullability(true)`:

```
------ Run test started ------
xUnit.net VSTest Adapter v2.4.1-pre.build.4059 (32-bit Desktop .NET 4.0.30319.42000)
  Starting:    Microsoft.CodeAnalysis.CSharp.Semantic.UnitTests
    Microsoft.CodeAnalysis.CSharp.UnitTests.Semantics.NullableReferenceTypesTests.Constraint_Oblivious_03 [FAIL]
      
      Expected:
                      Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(15, 9),
                      Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(4, 24),
                      Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(17, 9),
                      Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(19, 9),
                      Diagnostic(ErrorCode.WRN_NullabilityMismatchInTypeParameterConstraint, "G9").WithArguments("B4<T>", "A", "T", "A?").WithLocation(35, 12),
                      Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(21, 9),
                      Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(13, 9)
      Actual:
                      // (4,24): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' context.
                      // class B1<T> where T : A? { }
                      Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(4, 24),
                      // (15,9): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' context.
                      //     B1<A?> F3; // 2
                      Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(15, 9),
                      // (17,9): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' context.
                      //     B2<A?> F5; // 3
                      Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(17, 9),
                      // (19,9): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' context.
                      //     B3<A?> F7; // 4
                      Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(19, 9),
                      // (21,9): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' context.
                      //     B4<A?> F9; // 5 and 6
                      Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(21, 9),
                      // (13,9): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' context.
                      //     B0<A?> F1; // 1
                      Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(13, 9)
      Diff:
      ++>                 Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(4, 24)
                          Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(15, 9)
      -->                 Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(4, 24)
                          Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(17, 9)
                          Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(19, 9)
      -->                 Diagnostic(ErrorCode.WRN_NullabilityMismatchInTypeParameterConstraint, "G9").WithArguments("B4<T>", "A", "T", "A?").WithLocation(35, 12)
                          Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(21, 9)
                          Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(13, 9)
      Expected: True
      Actual:   False
      Stack Trace:
        src\Test\Utilities\Portable\Diagnostics\DiagnosticExtensions.cs(85,0): at Microsoft.CodeAnalysis.DiagnosticExtensions.Verify(IEnumerable`1 actual, DiagnosticDescription[] expected, Boolean errorCodeOnly)
        src\Test\Utilities\Portable\Diagnostics\DiagnosticExtensions.cs(47,0): at Microsoft.CodeAnalysis.DiagnosticExtensions.Verify(IEnumerable`1 actual, DiagnosticDescription[] expected)
        src\Compilers\CSharp\Test\Semantic\Semantics\NullableReferenceTypesTests.cs(53140,0): at Microsoft.CodeAnalysis.CSharp.UnitTests.Semantics.NullableReferenceTypesTests.Constraint_Oblivious_03()
    Microsoft.CodeAnalysis.CSharp.UnitTests.Semantics.NullableReferenceTypesTests.Constraint_Oblivious_04 [FAIL]
      
      Expected:
                      Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(4, 31),
                      Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(15, 16),
                      Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(17, 16),
                      Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(19, 16),
                      Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(21, 16),
                      Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(13, 16),
                      Diagnostic(ErrorCode.WRN_NullabilityMismatchInTypeParameterConstraint, "G4").WithArguments("B1<T>", "A<object?>", "T", "A<object>").WithLocation(30, 19),
                      Diagnostic(ErrorCode.WRN_NullabilityMismatchInTypeParameterConstraint, "G8").WithArguments("B3<T>", "A<object?>", "T", "A<object>").WithLocation(34, 19),
                      Diagnostic(ErrorCode.WRN_NullabilityMismatchInTypeParameterConstraint, "G9").WithArguments("B4<T>", "A<object>", "T", "A<object?>").WithLocation(35, 20)
      Actual:
                      // (4,31): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' context.
                      // class B1<T> where T : A<object?> { }
                      Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(4, 31),
                      // (15,16): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' context.
                      //     B1<A<object?>> F3; // 2
                      Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(15, 16),
                      // (17,16): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' context.
                      //     B2<A<object?>> F5; // 3
                      Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(17, 16),
                      // (19,16): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' context.
                      //     B3<A<object?>> F7; // 4
                      Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(19, 16),
                      // (21,16): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' context.
                      //     B4<A<object?>> F9; // 5 and 6
                      Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(21, 16),
                      // (13,16): warning CS8632: The annotation for nullable reference types should only be used in code within a '#nullable' context.
                      //     B0<A<object?>> F1; // 1
                      Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(13, 16)
      Diff:
                          Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(4, 31)
                          Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(15, 16)
                          Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(17, 16)
                          Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(19, 16)
                          Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(21, 16)
                          Diagnostic(ErrorCode.WRN_MissingNonNullTypesContextForAnnotation, "?").WithLocation(13, 16)
      -->                 Diagnostic(ErrorCode.WRN_NullabilityMismatchInTypeParameterConstraint, "G4").WithArguments("B1<T>", "A<object?>", "T", "A<object>").WithLocation(30, 19)
      -->                 Diagnostic(ErrorCode.WRN_NullabilityMismatchInTypeParameterConstraint, "G8").WithArguments("B3<T>", "A<object?>", "T", "A<object>").WithLocation(34, 19)
      -->                 Diagnostic(ErrorCode.WRN_NullabilityMismatchInTypeParameterConstraint, "G9").WithArguments("B4<T>", "A<object>", "T", "A<object?>").WithLocation(35, 20)
      Expected: True
      Actual:   False
      Stack Trace:
        src\Test\Utilities\Portable\Diagnostics\DiagnosticExtensions.cs(85,0): at Microsoft.CodeAnalysis.DiagnosticExtensions.Verify(IEnumerable`1 actual, DiagnosticDescription[] expected, Boolean errorCodeOnly)
        src\Test\Utilities\Portable\Diagnostics\DiagnosticExtensions.cs(47,0): at Microsoft.CodeAnalysis.DiagnosticExtensions.Verify(IEnumerable`1 actual, DiagnosticDescription[] expected)
        src\Compilers\CSharp\Test\Semantic\Semantics\NullableReferenceTypesTests.cs(53214,0): at Microsoft.CodeAnalysis.CSharp.UnitTests.Semantics.NullableReferenceTypesTests.Constraint_Oblivious_04()
  Finished:    Microsoft.CodeAnalysis.CSharp.Semantic.UnitTests
========== Run test finished: 2 run (0:00:03.1275898) ==========
```